### PR TITLE
Fix `extraConfigs` range operator

### PR DESCRIPTION
**Pull request by tuladhar**
This PR:

- Fixes the following issue with `extraConfigs` range operator

In values.yaml
```
  teleportKubeAgent:
    ...
    extraConfigs:
    - kind: configMap
      name: "{{ $.Values.clusterName }}-teleport-kube-agent-config"
```

Helm template results:
```
> helm template helm/default-apps-cloud-director/ -f helm/default-apps-cloud-director/ci/ci-values.yaml
Error: YAML parse error on default-apps-cloud-director/templates/apps.yaml: error converting YAML to JSON: yaml: line 39: did not find expected key

Use --debug flag to render out invalid YAML
```

### Testing

Description on how default-apps-cloud-director can be tested.

- [ ] fresh install works
- [ ] upgrade from previous version works

### Checklist

- [x] Update changelog in CHANGELOG.md, **check all commits are in it before release (renovate included)**.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/default-apps-cloud-director/templates/apps.yaml [hunk 1]
```diff
@@ -52,9 +52,9 @@ spec:
 {{- end }}
 {{- if .extraConfigs }}
   extraConfigs:
-  {{- range .extraConfigs }}
-  - kind: {{ .kind }}
-    name: {{ .name }}
+  {{- range $extraConfig := .extraConfigs }}
```
Comment (erkanerol) [inline helm/default-apps-cloud-director/templates/apps.yaml:55]: @tuladhar Does this work when `.name` is just a string too?

Comment (tuladhar) [reply helm/default-apps-cloud-director/templates/apps.yaml:55]: Yes, it does.
```
 (teleport)> yq '.apps.teleportKubeAgent.extraConfigs' helm/default-apps-cloud-director/values.yaml
- kind: configMap
  name: "bigmac-teleport-kube-agent-config"
```
```
 (teleport)> helm template helm/default-apps-cloud-director/ -f helm/default-apps-cloud-director/ci/ci-values.yaml | grep extraConfigs -A3
  extraConfigs:
  - kind: configMap
    name: bigmac-teleport-kube-agent-config
    namespace: default
```

```
 (teleport)> cat helm/default-apps-cloud-director/templates/apps.yaml | grep "range \$extraConfig" -A5
  {{- range $extraConfig := .extraConfigs }}
  - kind: {{ $extraConfig.kind }}
    name: {{ tpl $extraConfig.name $ }}
    namespace: {{ $.Release.Namespace }}
  {{- end }}
 {{- end }}
```

Comment (erkanerol) [reply helm/default-apps-cloud-director/templates/apps.yaml:55]: Then it is fine.

**Comment by tuladhar** [comment]
`CAPVCD Upgrade Suite` is expected to fail, see: https://gigantic.slack.com/archives/C04KA9DBMB2/p1697552000226339

```
Running Suite: CAPVCD Upgrade Suite - /app/providers/capvcd/upgrade
===================================================================
Random Seed: 1697702330

Will run 17 of 17 specs
------------------------------
[BeforeSuite] 
/app/providers/capvcd/upgrade/capvcd_suite_test.go:30
  {"level":"info","ts":"2023-10-19T07:58:50Z","msg":"Workload cluster name: t-nof46sr"}
  [FAILED] in [BeforeSuite] - /app/providers/capvcd/upgrade/capvcd_suite_test.go:69 @ 10/19/23 07:58:50.604
[BeforeSuite] [FAILED] [0.065 seconds]
[BeforeSuite] 
/app/providers/capvcd/upgrade/capvcd_suite_test.go:30

  [FAILED] Unexpected error:
      <*url.Error | 0xc0002abb30>: 
      Get "https://178.170.32.23:6443/api?timeout=32s": Service Unavailable
      {
          Op: "Get",
          URL: "https://178.170.32.23:6443/api?timeout=32s",
          Err: <*errors.errorString | 0xc00049a1f0>{
              s: "Service Unavailable",
          },
      }
  occurred
  In [BeforeSuite] at: /app/providers/capvcd/upgrade/capvcd_suite_test.go:69 @ 10/19/23 07:58:50.604
```

**Pull request merged**